build.assets/Dockerfile-centos7

```diff
@@ -176,6 +176,23 @@ ENV GOPATH="/go" \
     GOROOT="/opt/go" \
     PATH="/opt/bin:$PATH:/opt/go/bin:/go/bin:/go/src/github.com/gravitational/teleport/build"
 
+# BUILDARCH is automatically set by DOCKER when building the image with Build Kit (MacOS by deafult).
```

Comment:
This works on Linux but only when the build kit is enabled (it's off by default). Otherwise the compilation fails:
```
error: failed to run custom build command for `role_tester v0.1.0 (/go/src/github.com/gravitational/teleport/lib/datalog/roletester)`

Caused by:
  process didn't exit successfully: `/go/src/github.com/gravitational/teleport/target/release/build/role_tester-9a6781faf8447344/build-script-build` (exit status: 1)
  --- stderr
  Error: Custom { kind: Other, error: "protoc failed: /usr/local/bin/protoc: /usr/local/bin/protoc: cannot execute binary file\n" }
make[3]: *** [roletester] Error 101
```

Any reason why we cannot pass that variable in the same way as we're doing this currently for Ubuntu??
https://github.com/gravitational/teleport/blob/master/build.assets/Makefile#L127

Reply:
No. We should do it just like we do for Ubuntu. I didn't see that we were setting it explicitly.